### PR TITLE
Add ruff `RUF` rules

### PR DESCRIPTION
**Pull request by juanitorduz**
Add `RUF` rules as the PyMC configuration.

----
📚 Documentation preview 📚: https://pymc-marketing--636.org.readthedocs.build/en/636/

### CODE REVIEW

File: pymc_marketing/mmm/base.py [hunk 1]
```diff
@@ -665,7 +665,7 @@ def plot_budget_scenearios(
         standardize_scenarios_dict_keys(base_data, ["contribution", "budget"])
 
         fig, axes = plt.subplots(nrows=1, ncols=2, figsize=(15, 6))
-        scenarios = [base_data] + list(scenarios_data)
+        scenarios = [base_data, *list(scenarios_data)]
```
Comment (wd60622) [inline pymc_marketing/mmm/base.py:668]: Was this whole method not tested to begin with?

**Review by cetagostini — APPROVED** [review]

File: pyproject.toml [hunk 2]
```diff
@@ -59,7 +59,12 @@ repository = "https://github.com/pymc-labs/pymc-marketing"
 #changelog = ""
 
 [tool.ruff.lint]
-select = ["E", "F", "I", "W", "UP"]
+select = ["E", "F", "I", "W", "UP", "RUF"]
+ignore = [
+    "RUF001", # String contains ambiguous character (such as Greek letters)
+    "RUF002", # Docstring contains ambiguous character (such as Greek letters)
+    "RUF012", # Mutable class attributes should be annotated with `typing.ClassVar`
+]
```
Comment (wd60622) [inline pyproject.toml:67]: For the test suite, you might want you use something file specific ruff rules. For instance, the assert statements

```toml
[tool.ruff.lint.per-file-ignores]
"__init__.py" = ["F401"]
"**/tests/*" = ["S101"]

File: pymc_marketing/mmm/base.py (hunk 1 above)
Comment (ricardoV94) [reply pymc_marketing/mmm/base.py:668]: Why? that's equivalent code

Comment (juanitorduz) [reply pymc_marketing/mmm/base.py:668]: This was an automatic ruff refactor 😄

File: pyproject.toml (hunk 2 above)
Comment (juanitorduz) [reply pyproject.toml:67]: Ok! Actually these ignores I took from the PyMC config as they appear on the code base. I'll create a tickett to improve the rules in the test suite!

File: pymc_marketing/mmm/base.py (hunk 1 above)
Comment (wd60622) [reply pymc_marketing/mmm/base.py:668]: Says there is a code coverage difference

**Review by wd60622 — APPROVED** [review]
Looks good!

File: pymc_marketing/mmm/base.py (hunk 1 above)
Comment (ricardoV94) [reply pymc_marketing/mmm/base.py:668]: Ah... maybe it's only used in NB plots?

Comment (juanitorduz) [reply pymc_marketing/mmm/base.py:668]: I think so :)

**Pull request merged**